File: apps/desktop/src/stores/goals.ts
import { create } from 'zustand';

import type { GoalEntry } from '../lib/tauri-api';
import { tauriApi } from '../lib/tauri-api';

/**
 * «Прогресс целей» (#35, vision-волна 2): vault-широкий список заметок-целей (инлайн-тег `#goal`) с
 * прогрессом из frontmatter `progress`. Чистый SQL-read (офлайн, без LLM). Живой пересчёт по любому
 * modify goal-файла требует event-канала индексатора (см. BACKLOG) — в v1 обновление по `load`
 * (открытие панели + кнопка «Обновить»).
 */
interface GoalsState {
  items: GoalEntry[];
  loading: boolean;
  load: () => Promise<void>;
}

export const useGoalsStore = create<GoalsState>((set) => ({
  items: [],
  loading: false,
  async load() {
    set({ loading: true });
    try {
      const items = await tauriApi.goals.list();
      set({ items, loading: false });
    } catch {
      set({ items: [], loading: false });
    }
  },
}));

File: apps/desktop/src-tauri/src/commands/goals.rs
//! Команда «Прогресс целей» (#35, vision-волна 2): кросс-файловый список заметок-целей (#goal).

use tauri::State;

use crate::error::AppResult;
use crate::goals::{self, Goal};
use crate::state::AppState;

/// Все заметки-цели (инлайн-тег `#goal`) с прогрессом 0–100 (`null` — нет валидного значения, D7).
/// Без открытого vault — ошибка. Чистый SQL-read (офлайн, без LLM/сети).
#[tauri::command]
pub async fn list_goals(state: State<'_, AppState>) -> AppResult<Vec<Goal>> {
    let reader = state.vault().await?.db.reader().clone();
    Ok(goals::list_goals(&reader).await?)
}
